packages/std/src/query/mod.rs

```diff
@@ -5,6 +5,25 @@ use serde::{Deserialize, Serialize};
 use crate::Binary;
 use crate::Empty;
 
+/// Implements a hidden constructor for query responses.
+macro_rules! impl_response_constructor {
+    ( $response:ty, $( $field: ident : $t: ty),* ) => {
+        impl $response {
+            /// Constructor for testing frameworks such as cw-multi-test.
+            /// This is required because query response types should be #[non_exhaustive].
+            /// As a contract developer you should not need this constructor since
+            /// query responses are constructed for you via deserialization.
+            ///
+            /// Warning: This can change in breaking ways in minor versions.
+            #[doc(hidden)]
+            #[allow(dead_code)]
+            pub fn new($( $field: $t),*) -> Self {
+                Self { $( $field: $field.into() ),* }
```

Comment:
With this into() the constructor arg can have a different type than the field itself? Looks cool. Do you have an example for where they are different?

Reply:
Ah, that's actually a leftover from a version where I set all the parameters to be `impl Into<$t>`.
Keeping the `.into()` here would allow making some of the arguments `impl Into<_>` on the caller-side, but not sure if we really need that, since these constructors are mostly for internal use anyways?
I haven't done it in any of the calls here, but e.g. the `BondedDenomResponse` could have this one:
```rust
impl_response_constructor!(BondedDenomResponse, denom: impl Into<String>);
```

Reply:
Then let's simplify and remove the `.into()`. We can always add it back if needed.